src/qwed_new/core/cache.py

```
@@ -162,150 +224,248 @@ def stats(self) -> Dict[str, Any]:
-        
+
         try:
             return self._client.delete(key) > 0
-        except Exception:
-            return False
-    
+        except Exception as exc:
+            self._handle_runtime_redis_error("invalidate", exc)
+            return self._fallback_cache.invalidate(dsl_code, variables)
+
     def clear(self, pattern: str = None) -> int:
```

Ruff (0.15.14)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)